### PR TITLE
Migrate parsing to YARP

### PR DESCRIPTION
**Pull request by Morriar**
I've been experimenting with different parser backends to benchmark RBI parsing.

Here are the time required to parse all the 11801 RBI files in on big project:

| Backend | Parsing Time |
| - | :-: |
| Whitequark   | 288s |
| SyntaxTree   | 548s |
| YARP         | 22s |

This PR migrates RBI parsing to YARP and removes the dependecy to Whitequark and `unparser`.

Running `tapioca check-shims` before (using `parser` gem):

```
Loading Sorbet payload...  Done (3.72s)
Loading shim RBIs from sorbet/rbi/shims... Done (0.97s)
Loading gem RBIs from sorbet/rbi/gems...  Done (204.15s)
Loading dsl RBIs from sorbet/rbi/dsl...  Done (69.79s)
Loading annotation RBIs from sorbet/rbi/annotations...  Done (1.04s)
Looking for duplicates...  Done (0.56s)

No duplicates found in shim RBIs

real	4m43.293s
```

Running `tapioca check-shims` after (using YARP):

```
Loading Sorbet payload...  Done (3.09s)
Loading shim RBIs from sorbet/rbi/shims... Done (2.28s)
Loading gem RBIs from sorbet/rbi/gems...  Done (18.49s)
Loading dsl RBIs from sorbet/rbi/dsl...  Done (21.41s)
Loading annotation RBIs from sorbet/rbi/annotations...  Done (0.51s)
Looking for duplicates...  Done (0.52s)

No duplicates found in shim RBIs

real	0m56.171s
```

### CODE REVIEW

File: lib/rbi/loc.rb [hunk 1]
```diff
@@ -5,6 +5,21 @@ module RBI
...
+      extend T::Sig
+
+      sig { params(file: String, yarp_location: YARP::Location).returns(Loc) }
+      def from_yarp(file, yarp_location)
+        new(
+          file: file,
+          begin_line: yarp_location.start_line,
+          end_line: yarp_location.end_line,
+          begin_column: yarp_location.start_column,
+          end_column: yarp_location.end_column + 1, # TODO: Why is this off by one?
```
Comment (andyw8) [inline lib/rbi/loc.rb:18]: Known issue, I'm verifying Kevin's suggestion for a fix [here](https://github.com/ruby/yarp/compare/main...andyw8:yarp:andyw8/adjust-column-behaviour).

**Comment by Morriar** [comment]
I bumped YARP to 0.0.10 and tested the changes on core 👍

File: lib/rbi/loc.rb (hunk 1 above)
Comment (Morriar) [reply lib/rbi/loc.rb:18]: Great! I'm going to merge and update this once fixed 👍

File: sorbet/rbi/shims/gems/yarp.rbi [hunk 2]
```diff
@@ -0,0 +1,216 @@
+# typed: strict
```
Comment (andyw8) [inline sorbet/rbi/shims/gems/yarp.rbi:1]: You might want to use the RBI @vinistock added in https://github.com/Shopify/ruby-lsp/pull/978

Comment (Morriar) [reply sorbet/rbi/shims/gems/yarp.rbi:1]: Without more automation this is going to go out of date pretty quickly

**Pull request merged**